# Домашнее задание 4

## Задача 1

О непрерывной равномерно распределенной случайной величине B известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

#### Решение

$\xi $ ~ $R[0.5, b]$, $E\xi = \int\limits_a^b p_{\xi}(x) dx = \int\limits_a^b \frac{x - a}{b - a} dx = \frac{a+b}{2}, D\xi = \frac{(b - a)^2}{12}$

$D \xi = 0.2 = \frac{(b - 0.5)^2}{12}$


$(b - 0.5)^2 = 2.4$

$b - 0.5 = \sqrt{2.4} \Leftrightarrow b = \sqrt{2.4} + 0.5 = 2.05$

In [3]:
2.4**(0.5) + 0.5

2.049193338482967

$E\xi = \frac{2.05 + 0.5}{2} = 1.275$

In [4]:
(2.05+ 0.5)/2

1.275

## Задача 2

Рост взрослого населения города X имеет нормальное распределение, причем, средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. посчитайте, какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
1. больше 182 см?
2. больше 190 см?
3. от 166 см до 190 см?
4. от 166 см до 182 см?
5. от 158 см до 190 см?
6. не выше 150 см или не ниже 190 см?
7. не выше 150 см или не ниже 198 см?
8. ниже 166 см?
Задачу можно решить двумя способами: без использования сторонних библиотек (numpy, scipy, pandas и пр.), а затем проверить себя с помощью встроенных функций

#### Решение

Функция распределения $N(174, 8^2)$:

$F_\xi(x) = \frac{1}{\sqrt{2\pi}\sigma} \int \limits_{-\infty}^x e^{ -\frac{1}{2}\frac{(t - \mu)^2}{\sigma^2} }dt = \frac{1}{\sqrt{2\pi}\cdot8} \int \limits_{-\infty}^x e^{ -\frac{1}{2}\frac{(t - 174)^2}{8^2} }dt$

Попробуем интегрировать прямо в питоне, раньше этого не делал

In [5]:
import scipy.integrate as integrate
import numpy as np

In [26]:
def f_ksi(x, mu, sigma): # Плотность
    res = (1./ ((2*np.pi)**(0.5) * sigma) * np.exp(-0.5 * (x - mu)**2 / (sigma)**2))
    return res

def F_ksi(x, mu, sigma): #Функция распределения
    func = lambda t: f_ksi(t, mu, sigma)
#    result = integrate.quad(f_ksi(t, mu, sigma), )
    result = integrate.quad(func, -np.inf, x)[0]
    return result

### Ниже идет вспомогательный код вычислений, предлагаю листать ниже, ход мыслей все равно прописан

In [30]:
1 - F_ksi(182, 174, 8)

0.15865525393145719

In [47]:
1 - F_ksi(190, 174, 8)

0.02275013194817832

In [48]:
(1 - F_ksi(166, 174, 8)) *F_ksi(190, 174, 8)

0.8222040420815823

In [49]:
(1 - F_ksi(166, 174, 8)) *F_ksi(182, 174, 8)

0.7078609817371455

In [50]:
(1 - F_ksi(158, 174, 8)) *F_ksi(190, 174, 8)

0.955017304607302

In [51]:
(F_ksi(150, 174, 8)) + (1 - F_ksi(190, 174, 8))

0.024100029979801926

In [52]:
(F_ksi(150, 174, 8)) + (1 - F_ksi(198, 174, 8))

0.0026997960632537087

In [53]:
(F_ksi(166, 174, 8))

0.15865525393145166

In [33]:
from IPython.display import Markdown as md

In [45]:
#md("1. больше 182 см: 1 - $F_{\xi}(182)$ = {}".format(1 - F_ksi(182, 174, 8)))

Не разобрался пока, как правильно в такой строке пользоваться математическим текстом, так что пока по старинке

### А вот и ответы

1. больше 182 см?

$$1 - F_{\xi}(182) = 0.1586$$

2. больше 190 см?

$$1 - F_{\xi}(190) = 0.02$$

3. от 166 см до 190 см?

$$\big(1 - F_{\xi}(166)\big)\cdot F_{\xi}(190) = 0.82$$
4. от 166 см до 182 см?

$$\big(1 - F_{\xi}(166)\big)\cdot F_{\xi}(182) = 0.70$$
5. от 158 см до 190 см?

$$\big(1 - F_{\xi}(158)\big)\cdot F_{\xi}(190) = 0.955$$
6. не выше 150 см или не ниже 190 см?

$$F_{\xi}(150) + \big(1 - F_{\xi}(190)\big) = 0.024$$
7. не выше 150 см или не ниже 198 см?

$$F_{\xi}(150) + \big(1 - F_{\xi}(198)\big) = 0.0026$$
8. ниже 166 см?

$$F_{\xi}(166) = 0.158$$

## Задача 3

Коробки с шоколадом упаковываются автоматически. Их средняя масса равна 1.06 кг. Известно, что 5% коробок имеют массу, меньшую 1 кг. Найдите: а) среднее квадратическое отклонение массы коробки, б) процент коробок, имеющих массу больше 1.1 кг.
Подсказка. 1. Можно считать, что распределение массы коробки нормальное. 2. Найдите такое значение scale, для которого значение cdf(x=1, loc=1.06, scale=scale) близко к 0.05. Точности 0.0001 будет достаточно.

#### Решение

Аналогично 1-й задаче - знаем, мю, раз это нормальное распределение, то, зная функцию распределения, найдем соответствующую сигму

$P(\xi \le 1 кг) = F_{\xi}(1 кг) = 0.05$

Не пойму пока, как это решать по-человечески...попробуем "по-математически"

$Ф_{\mu, \sigma^2}(x) = Ф_{0, 1}(\frac{x - \mu}{\sigma})$

Воспользуемся таблицей, найденной на просторах интернета

 Как я понял, где-то около -1.65: 

$\frac{1 - 1.06}{\sigma} = -1.65$

$\sigma = 1.65\cdot 0.06 = 0.09$

In [60]:
1.65*0.06

0.09899999999999999

__Ответ__

90 грам - среднеквадратичное отклонение

Теперь, когда оценили std, найдем $1 - F_{\xi}(1.1)$

In [55]:
from scipy import stats

In [62]:
1 - stats.norm.cdf( 1.1, loc = 1.06, scale = 0.09)

0.3283606432818851

__Ответ:__

32% коробок

In [58]:
?stats.norm.ppf()

Signature: stats.norm.ppf(q, *args, **kwds)
Docstring:
Percent point function (inverse of `cdf`) at q of the given RV.

Parameters
----------
q : array_like
    lower tail probability
arg1, arg2, arg3,... : array_like
    The shape parameter(s) for the distribution (see docstring of the
    instance object for more information)
loc : array_like, optional
    location parameter (default=0)
scale : array_like, optional
    scale parameter (default=1)

Returns
-------
x : array_like
    quantile corresponding to the lower tail probability q.
File:      /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/scipy/stats/_distn_infrastructure.py
Type:      method


In [59]:
stats.norm.ppf(0.05)

-1.6448536269514729